azure-key-vault/operations.py

```diff
@@ -0,0 +1,352 @@
+        except requests.exceptions.ConnectionError:
+            logger.error('A connection error occurred')
+            raise ConnectorError('A connection error occurred')
+        except requests.exceptions.Timeout:
+            logger.error('The request timed out')
+            raise ConnectorError('The request timed out')
+        except requests.exceptions.RequestException:
+            logger.error('There was an error while handling the request')
+            raise ConnectorError('There was an error while handling the request')
+        except Exception as e:
+            logger.error('{0}'.format(e))
+            raise ConnectorError('{0}'.format(e))
+
+    def make_rest_call_vault(self, endpoint, params={}, data=None, method='POST'):
+        headers = {'Authorization': self.vault_token, 'Content-Type': 'application/json'}
+        service_url = self.server_url + endpoint
+        logger.debug('Request URL {}'.format(service_url))
+        params['api-version'] = VAULT_API_VERSION
+        try:
+            if data:
+                data = json.dumps(data)
+            response = requests.request(method, service_url, data=data, headers=headers, params=params,
+                                        verify=self.verify_ssl)
+            if response.ok:
+                content_type = response.headers.get('Content-Type')
+                if response.text != "" and 'application/json' in content_type:
+                    return response.json()
+                else:
+                    return response.content
+            else:
+                if response.text != "":
+                    err_resp = json.loads(response.json()) if type(response.json()) is str else response.json()
+                    if "error" in err_resp:
+                        error_msg = "{}: {}".format(err_resp.get('error').get('code'),
+                                                    err_resp.get('error').get('message'))
+                        raise ConnectorError(error_msg)
+                else:
+                    error_msg = '{0}: {1}'.format(response.status_code, response.reason)
+                    raise ConnectorError(error_msg)
+        except requests.exceptions.SSLError:
+            logger.error('An SSL error occurred')
+            raise ConnectorError('An SSL error occurred')
+        except requests.exceptions.ConnectionError:
+            logger.error('A connection error occurred')
+            raise ConnectorError('A connection error occurred')
+        except requests.exceptions.Timeout:
+            logger.error('The request timed out')
+            raise ConnectorError('The request timed out')
+        except requests.exceptions.RequestException:
+            logger.error('There was an error while handling the request')
+            raise ConnectorError('There was an error while handling the request')
+        except Exception as e:
+            logger.error('{0}'.format(e))
+            raise ConnectorError('{0}'.format(e))
```

Comment:
The code is almost identical to make_rest_call. can we merge the both functions.

Reply:
yes, merged